enumerables.rb

```diff
@@ -0,0 +1,166 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |x|
+      yield(self[x])
+    end
+    self
+  end
+
+  def my_each_with_index(value = 0)
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = value
+    size.times do |x|
+      yield(self[x], index)
+      index += 1
+    end
+  end
+
+  def my_select(arg = nil)
+    return to_enum(:my_select) unless block_given? || !arg.nil?
+
+    arr = []
+
+    if !arg.nil?
+      my_each do |x|
+        arr.push(x) if arg(x)
+      end
+    else
+      my_each do |x|
+        arr.push(x) if yield(x)
+      end
+    end
+    arr
+  end
```

Comment:
# my_select
1. [ ] Returns an array containing all elements of enum for which the given block returns a true value:: range
### Test 👉 `range.my_select(&block)`
Resulted in *NoMethodError*